### ISSUE TITLE
Metadata seems to be missing

### ISSUE THREAD

**Issue by Ilya-666** [PyO3/pyo3#4752]
### Bug Description

When I try to install it comes up as unknown.
 pip install git+https://github.com/PyO3/pyo3.git@v0.23.2

### Steps to Reproduce

pip install git+https://github.com/PyO3/pyo3.git@v0.23.2

### Backtrace

```shell
Collecting git+https://github.com/PyO3/pyo3.git@v0.23.2
  Cloning https://github.com/PyO3/pyo3.git (to revision v0.23.2) to /tmp/pip-req-build-s2iv13aq
  Running command git clone --filter=blob:none --quiet https://github.com/PyO3/pyo3.git /tmp/pip-req-build-s2iv13aq
  Running command git checkout -q a9678269d4742eb2cd5eda19d6935eaea44f693e
  Resolved https://github.com/PyO3/pyo3.git to commit a9678269d4742eb2cd5eda19d6935eaea44f693e
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: UNKNOWN
  Building wheel for UNKNOWN (pyproject.toml) ... done
  Created wheel for UNKNOWN: filename=UNKNOWN-0.0.0-py3-none-any.whl size=5649 sha256=08fdfea069c13d1752849a066acd828bb2e4759bfd2d6f723c0627eafe8451da
  Stored in directory: /tmp/pip-ephem-wheel-cache-j5t947jl/wheels/4e/cb/1e/0c0c9de8e26d14957e85ce2953049056abecd11392376ed5e4
Successfully built UNKNOWN
Installing collected packages: UNKNOWN
Successfully installed UNKNOWN-0.0.0
```

### Your operating system and version

ubuntu 24.04

### Your Python version (`python --version`)

3.13.0

### Your Rust version (`rustc --version`)

rustc 1.83.0 (90b35a623 2024-11-26)

### Your PyO3 version

23.2

### Additional Info

also happens if I install locally

**Comment by alex** [comment]
This isn't a Python package, it's not expected to be installed at all. pyo3 is used _by_ python packages, but it itself is a Rust package.

**Comment by Ilya-666** [comment]
Ok, I’m struggling to get transformers and qwen to use the latest version so I tried to install it directly after tweaking the toml for transformers.
Maybe am being thick but can’t get qwen and transformers to run on python 3.13, I thought the latest version would fix that.


Sent from Yahoo Mail for iPhone


On Monday, December 2, 2024, 1:17 pm, Lily Foote ***@***.***> wrote:




Closed #4752 as completed.

**Comment by davidhewitt** [comment]
For transformers, I think the dependency which would have PyO3 is `tokenizers`? You might be ok if you use a sufficiently new `tokenizers` release: https://github.com/huggingface/tokenizers/releases/tag/v0.20.2

Not sure about `qwen`, but the same applies; if you can figure out which dependency is using Rust/PyO3 then you can ask them to update to PyO3 0.22 for 3.13 (or 0.23 for 3.13t).
